File: client/src/app/modules/quotation/services/rejection.cause.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { RejectionCause } from '../../quotation/model/RejectionCause';

@Injectable({
  providedIn: 'root'
})
export class RejectionCauseService extends AppRestService<RejectionCause>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getRejectionCauses() {
    return this.getList(new HttpParams(), "rejection-causes");
  }
  
   addOrUpdateRejectionCause(rejectionCause: RejectionCause) {
    return this.addOrUpdate(new HttpParams(), "rejection-cause", rejectionCause, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/RejectionCauseService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.RejectionCause;

public interface RejectionCauseService {
    public List<RejectionCause> getRejectionCauses();

    public RejectionCause getRejectionCause(Integer id);

    public RejectionCause addOrUpdateRejectionCause(RejectionCause rejectionCause);
}
